bindings/rust/s2n-tls/src/enums.rs

```diff
@@ -31,6 +31,26 @@ impl<T, E> From<Result<T, E>> for CallbackResult {
     }
 }
 
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum FipsMode {
+    Disabled,
+    Enabled,
+}
```

Comment:
Can someone confirm that this should be non-exhaustive.

Reply:
@goatgoose confirmed that we might add some other states to this.

Reply:
Yeah - [this was discussed](https://github.com/aws/s2n-tls/pull/4450#discussion_r1523999520) when adding the FIPS getter. We decided to use an enum in case we want to add a new value in the future, rather than returning only true/false. So making this non-exhaustive makes sense.